geomesa-hbase/geomesa-hbase-datastore/src/main/java/org/locationtech/geomesa/hbase/filters/JSimpleFeatureFilter.java

```diff
@@ -57,18 +83,34 @@ public ReturnCode filterKeyValue(Cell v) throws IOException {
         if (filter == null) {
             return ReturnCode.INCLUDE;
         } else {
-            SimpleFeature sf = serializer.deserialize(v.getValueArray(), v.getValueOffset(), v.getValueLength());
-            if (filter.evaluate(sf)) {
-                return ReturnCode.INCLUDE;
-            } else {
+            try {
+                log.trace("cell = {}", v);
+                byte[] valueBytes = v.getValueArray();
+                int valueOffset = v.getValueOffset();
+                int valueLength = v.getValueLength();
+                log.trace("valueBytes.length = {}, valueOffset = {}, valueLength = {}, column = {}",
+                        valueBytes.length,
+                        valueOffset,
+                        valueLength,
+                        Bytes.toString(v.getFamilyArray(), v.getFamilyOffset(), v.getFamilyLength())
+                );
+                reusable.setBuffer(v.getValueArray(), v.getValueOffset(), v.getValueLength());
+                log.trace("Evaluating filter against SimpleFeature");
+                if (filter.evaluate(reusable)) {
+                    return ReturnCode.INCLUDE;
+                } else {
+                    return ReturnCode.SKIP;
+                }
+            } catch(Exception e) {
+                log.error("Exception thrown while scanning, skipping", e);
                 return ReturnCode.SKIP;
             }
         }
     }
 
     @Override
     public Cell transformCell(Cell v) throws IOException {
-        return super.transformCell(v);
```

Comment:
I think I saw this is other implementations of HBase Filters.  Maybe I'm imagining things again.  

The change is probably ok, but it might be worth thinking about.